packages/vector_graphics/lib/src/listener.dart

```diff
@@ -95,19 +117,26 @@ class FlutterVectorGraphicsListener extends VectorGraphicsCodecListener {
   final TextDirection? _textDirection;
 
   final ui.PictureRecorder _recorder;
-  final ui.Canvas _canvas;
+  ui.Canvas _canvas;
 
   final List<ui.Paint> _paints = <ui.Paint>[];
   final List<ui.Path> _paths = <ui.Path>[];
   final List<ui.Shader> _shaders = <ui.Shader>[];
   final List<_TextConfig> _textConfig = <_TextConfig>[];
   final List<Future<ui.Image>> _pendingImages = <Future<ui.Image>>[];
   final Map<int, ui.Image> _images = <int, ui.Image>{};
-
+  final List<Future<ui.ImageShader>> _pendingPatterns =
```

Comment:
Instead of adding a new list to tracking patterns here, lets combine this with the pending images check. if the list is typed as Future<void> it can accept either future.

```
final List<Future<void>> _pendingWork = <Future<void>>[];
```

Then you only need a single `waitFor_` method. Does that work?

Reply:
I realized that after toImageSync is used, _pendingPatterns is needed at all, nor is waitForPatternDecode(), so I removed both.

Reply:
perfect!